__tests__/app.test.js

```diff
@@ -140,72 +144,150 @@ describe("api/articles/:article_id/comments ", () => {
+                    return request(app)
+                    .get("/api/articles/1/comments")
+                    .expect(200)
+                    .then(({ body }) => {
+                        expect(body.comments).toBeSortedBy("created_at", {descending: true})
+                        })
+                        
+                    })
+        test("GET: 400 response when passed an article_id that is not a number", () => {
+                    return request(app)
+                        .get("/api/articles/not-a-number/comments")
+                        .expect(400)
+                        .then(({ body }) => {
+                            expect(body.msg).toBe("Bad Request!")
+                                })
+                                
+                    })
+        test("GET: 404 response when passed a number that does not exist on our table", () => {
+                    return request(app)
+                        .get("/api/articles/999/comments")
+                        .expect(404)
+                        .then(({ body }) => {
+                            expect(body.msg).toBe("Not Found!")
+                              })
+                                                
+                    })
+        test("GET: 200 response with an empty array if the article_id is valid but there are no comments on that article", () => {
+                    return request(app)
+                    .get("/api/articles/2/comments")
+                    .expect(200)
+                    .then(({ body }) => {
+                    expect(Array.isArray(body.comments)).toBe(true)
+                    expect(body.comments).toHaveLength(0)
+                                })
+                                
+                    })
             })
-         })
-    })
-    test("GET: 200 response with an array of comments from an article with a given article ID ordered by data created with the most recent first by default", () => {
-        return request(app)
-        .get("/api/articles/1/comments")
-        .expect(200)
-        .then(({ body }) => {
-            expect(body.comments).toBeSortedBy("created_at", {descending: true})
+         
+    describe("POST/api/articles/:article_id/comments ",()=>{
+
+        test("POST: 201 adds comment to comment table with correct keys of body and username", () => {
+            const newComment = {
+                author: "butter_bridge",
+                body: "This is a test comment."
+            }
+            return request(app)
+            .post("/api/articles/1/comments")
+            .send(newComment)
+            .expect(201)
+            .then(({ body }) => {
+                expect(body.comment).toHaveProperty("author"),
+                expect(body.comment).toHaveProperty("body"),
+                expect(body.comment).toHaveProperty("article_id"),
```

Comment:
We don't need these tests if we have the following assertions, as if the `body.comment` does not have the properties in `.toMatchObject()` it will fail

Reply:
I think the test on 205 is a more semantic and comprehensive test 😄